lib/internal/modules/esm/loader.js

```diff
@@ -73,7 +78,13 @@ class Loader {
   async import(specifier, parent) {
     const job = await this.getModuleJob(specifier, parent);
     const module = await job.run();
```

Comment:
Job.run is only called for top-level imports, so if you provide the warning in the implementation of `job.run` you shouldn't need to separately cache the errors with a weak map.

Reply:
i put it here to catch the entry file and dynamic imports, do you think i should go for all imports?

Reply:
@devsnek I don't think we should do all imports, but as far as I'm aware, `job.run` will be called iff it is an entry file or dynamic import?